overlord/snapstate/service_control.go

```diff
@@ -0,0 +1,138 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package snapstate
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/snapcore/snapd/overlord/snapstate/backend"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/snap"
+
+	tomb "gopkg.in/tomb.v2"
+)
+
+// ServiceCommand encapsulates a single service-related action (such as starting,
+// stopping or restarting) run against services of a given snap. The action is
+// run for services listed in names attribute, or for all services of the snap
+// if names is empty.
+type ServiceCommand struct {
+	SnapName   string                 `json:"snap-name"`
+	Action     string                 `json:"action"`
+	Flag       string                 `json:"flag,omitempty"`
```

Comment:
This is weird, a flag that is a string?

Reply:
Fair point. FWTW this is a placeholder for flags such as "enable", "disable", "reload" that depend on the action.

Reply:
Perhaps calling it a `SupportAction` would do, assuming you will have at most one under any circumstances.

Reply:
maybe ActionModifier

Reply:
Thank you, I like that name.